dvc/cli.py

```diff
@@ -117,6 +158,83 @@ def parse_args(self, args=None, namespace=None):
             self.error(msg % " ".join(argv), getattr(args, "func", None))
         return args
 
+    def _get_cmd_choices(self):
+        """Keep track of public dvc commands and hidden dvc commands.
+
+        Returns:
+            A dictionary with public dvc commands mapped to a list containing
+            dvc subcommands (if they exist).
+
+            A list containing hidden dvc commands to be ignored when finding
+            suggestions.
+        """
+        cmd_choices = {}
+        hidden_cmds = []
+
+        parser_actions = self._actions  # pylint: disable=protected-access
+        for parser_action in parser_actions:
+            if parser_action.dest == "help":
+                # treat -h, --help as command choices
+                for option in parser_action.option_strings:
+                    cmd_choices[option] = []
+            elif parser_action.dest == "cmd":
+                for cmd, subparser in parser_action.choices.items():
+                    if not subparser.add_help:
+                        hidden_cmds.append(cmd)
+                        continue
+                    cmd_choices[cmd] = []
+                    actions = (
+                        subparser._actions  # pylint: disable=protected-access
+                    )
+                    for action in actions:
+                        if not isinstance(action.choices, dict):
+                            # NOTE: we are only interested in subcommands
+                            continue
+                        cmd_choices[cmd].extend(action.choices.keys())
+
+        return cmd_choices, hidden_cmds
+
+    def _find_cmd_suggestions(self, args):
```

Comment:
It is really weird how we have this method and a function above with the same name, need better name for one or the other.